### PR TITLE
Is243/create storage service (1st draft)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- [x] creates service using new cookiecutter template (issue #245 )
- [x] Unifies versioning of the api: syncs API, server and client packages (issue #229 ) using .bumpversion
- [x] moves @mguidon prototype into simcore (issue #243)
    - still not fully functional

## Related issue number
connect to #243 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you design a new module, add your user to .github/CODEOWNERS

[waffle.io]:https://waffle.io/marketing-assets/documents/waffleio_cheatsheet_v1.pdf?utm_source=blog&utm_medium=cheatsheet-ctabutton&utm_campaign=cheatsheet

### CODE REVIEW

File: services/storage/src/simcore_service_storage/handlers.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    data2 = {
+        'filename' : "a.txt",
+        'version': '1.0',
+        'last_accessed' : 1234.2,
+        'owner' : 'c8da5f29-6906-4d0f-80b1-0dc643d6303d',
+        'storage_location' : 'simcore.s3'
+    }
+
+    #return files_schema.dump([db_data1, db_data1])
+    return [data1, data2]
```
[the remark this answers is not in the document]
Comment (mguidon) [reply services/storage/src/simcore_service_storage/handlers.py]: Yup. Just fake. Will work on it today

File: services/storage/src/simcore_service_storage/handlers.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    fileId = path_params['fileId']
+
+    data = {
+        'filename' : "a.txt",
+        'version': '1.0',
+        'last_accessed' : 1234.2,
+        'owner' : 'c8da5f29-6906-4d0f-80b1-0dc643d6303d',
+        'storage_location' : 'simcore.s3'
+    }
+    return data
```
[the remark this answers is not in the document]
Comment (mguidon) [reply services/storage/src/simcore_service_storage/handlers.py]: same same

File: packages/service-library/src/servicelib/openapi.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+"""
+from pathlib import Path
+
+import openapi_core
+import yaml
+from openapi_core.schema.exceptions import OpenAPIError, OpenAPIMappingError #
+from openapi_core.schema.specs.models import Spec
+
+# Supported version of openapi
+OAI_VERSION = '3.0.1'
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply packages/service-library/src/servicelib/openapi.py]: That is the document I followed... it is a patch version and according to their [website](https://github.com/OAI/OpenAPI-Specification ... ) : 
Patch-level fixes typos, clarifications, etc.
For the moment this number is just a placeholder but should be used to check against the specs

File: services/storage/tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+@pytest.fixture(scope='session')
+def python27_exec(osparc_simcore_root_dir, tmpdir_factory, here):
+    # Assumes already created with make .venv27
+    venv27 = osparc_simcore_root_dir / ".venv27"
+
+    if not venv27.exists():
+        # create its own virtualenv
+        venv27 = tmpdir_factory.mktemp("virtualenv") / ".venv27"
+        cmd = "virtualenv --python=python27 '%s'"%(venv27) # TODO: how to split in command safely?
```
Comment (mguidon) [inline services/storage/tests/conftest.py]:
```suggestion
        cmd = "virtualenv --python=python2 %s"%(venv27) # TODO: how to split in command safely?
```

File: services/storage/setup.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        '': [
+            'data/*.json',
+            'data/*.yml',
+            'data/*.yaml',
+            'oas3/*.yaml',
+            'oas3/*.yml',
+            ],
+    },
+    #data_files = list_datafiles_at(
+    #    "etc/", # Contain the configuration files for all the programs that run on your system.
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/storage/setup.py]: Agreed!

File: services/storage/src/simcore_service_storage/datcore_wrapper.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+FileMetaDataVec = List[FileMetaData]
+
+CURRENT_DIR = Path(__file__).resolve().parent
+
+def call_python_2(module, function, args, python_exec: Path):
+    """ calls a module::function from python2 with the arguments list
+    """
+    # pylint: disable=E1101
+    # "E1101:Module 'execnet' has no 'makegateway' member",
+    gw = execnet.makegateway("popen//python=%s" % python_exec)
```
Comment (oetiker) [inline services/storage/src/simcore_service_storage/datcore_wrapper.py]: that does not look as if it was async ...

File: services/storage/src/simcore_service_storage/datcore_wrapper.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+        MaG: I wonder how secure it is to pass the tokens that way...
+
+    """
+    prefix = "import sys\n" \
+             "sys.path.append('%s')\n" % CURRENT_DIR
+    script = prefix + dedent(script)
+
+    # pylint: disable=E1101
+    gw = execnet.makegateway("popen//python=%s" % python_exec)
```
Comment (oetiker) [inline services/storage/src/simcore_service_storage/datcore_wrapper.py]: this does not look like it was async

**Pull request merged**